iam.tf

```diff
@@ -75,6 +75,12 @@ resource "aws_iam_instance_profile" "this" {
   tags = var.additional_tags
 }
 
+data "aws_kms_key" "secure_env_vars" {
+  count = local.has_secure_env_vars && var.create_iam_role ? 1 : 0
```

Comment:
Could you maybe extract the conditional to a var, so it remains the same for the two resources? Plus, it is easier to know what the combo means. Like:

```
locals {
  use_secure_env_vars = local.has_secure_env_vars && var.create_iam_role
}
```

Reply:
Done, also used the same local value in other places with the same logic.